src/main/java/hudson/plugins/promoted_builds/conditions/ManualCondition.java

```diff
@@ -239,6 +239,14 @@ public String getUserName() {
         }
 
         @Exported
+        public String getUserId() {
+            if (authenticationName == null)
```

Comment:
Thanks for pointing it!
Actually this method has been loaned from the original PR. authenticationName in getUserId() seems to be an issue, I need to review it